climakitae/threshold_tools.py

```diff
@@ -400,77 +405,90 @@ def get_return_value(
+                arg_value=arg_value,
             )
         except (ValueError, ZeroDivisionError):
-            return_value = np.nan
+            return_variable = np.nan
 
         conf_int_lower_limit, conf_int_upper_limit = _conf_int(
             ams=ams,
             distr=distr,
             data_variable=data_variable,
-            arg_value=return_period,
+            arg_value=arg_value,
             bootstrap_runs=bootstrap_runs,
             conf_int_lower_bound=conf_int_lower_bound,
             conf_int_upper_bound=conf_int_upper_bound,
         )
 
-        return return_value, conf_int_lower_limit, conf_int_upper_limit
+        return return_variable, conf_int_lower_limit, conf_int_upper_limit
 
-    return_value, conf_int_lower_limit, conf_int_upper_limit = xr.apply_ufunc(
-        return_value,
+    return_variable, conf_int_lower_limit, conf_int_upper_limit = xr.apply_ufunc(
+        _return_variable,
         ams,
         input_core_dims=[["time"]],
         exclude_dims=set(("time",)),
         output_core_dims=[[], [], []],
     )
 
-    return_value = return_value.rename("return_value")
-    new_ds = return_value.to_dataset()
+    return_variable = return_variable.rename(data_variable)
+    new_ds = return_variable.to_dataset()
     new_ds["conf_int_lower_limit"] = conf_int_lower_limit
     new_ds["conf_int_upper_limit"] = conf_int_upper_limit
 
     if multiple_points:
         new_ds = new_ds.unstack("allpoints")
 
-    new_ds["return_value"].attrs["return period"] = "1-in-{}-year event".format(
-        str(return_period)
-    )
+    new_ds.attrs = ams_attributes
+
+    if data_variable == "return_value":
+        new_ds["return_value"].attrs["return period"] = f"1-in-{arg_value}-year event"
+    elif data_variable == "return_prob":
+        threshold_unit = ams_attributes["units"]
+        new_ds["return_prob"].attrs[
+            "threshold"
+        ] = f"exceedance of {arg_value} {threshold_unit} event"
+        new_ds["return_prob"].attrs["units"] = None
+    elif data_variable == "return_period":
+        return_value_unit = ams_attributes["units"]
+        new_ds["return_period"].attrs[
+            "return value"
+        ] = f"{arg_value} {return_value_unit} event"
+        new_ds["return_period"].attrs["units"] = "years"
+
     new_ds["conf_int_lower_limit"].attrs[
         "confidence interval lower bound"
     ] = "{}th percentile".format(str(conf_int_lower_bound))
     new_ds["conf_int_upper_limit"].attrs[
         "confidence interval upper bound"
     ] = "{}th percentile".format(str(conf_int_upper_bound))
 
-    new_ds.attrs = ams_attributes
-    new_ds.attrs["distribution"] = "{}".format(str(distr))
+    new_ds.attrs["distribution"] = f"{distr}"
     return new_ds
 
 
-def get_return_prob(
+def get_return_value(
     ams,
-    threshold,
+    return_period=10,
     distr="gev",
     bootstrap_runs=100,
     conf_int_lower_bound=2.5,
     conf_int_upper_bound=97.5,
     multiple_points=True,
 ):
```

Comment:
I don't really see the purpose of this function since it's essentially just a copy of `_get_return_variable`, wrapped in another function with a different name. Why not just rename `_get_return_variable` to `get_return_value`, and have that user-facing function called in `get_return_prob` and `get_return_period`?

Reply:
These are the main functions called to run the code and have specific parameters needed for each. They now call a generaic internal function `_get_return_variable`  to reduce redundant code.

Reply:
I see your point, but `get_return_value` and `_get_return_variable` have the exact same argument defaults. If they had different default arguments I could maybe see having the two, but as they are now, I still think it makes more sens to just combine them into one. Any thoughts @ckingdon95 ? Definitely open to being wrong about this, haha

Reply:
Maybe this is a best practice python thing that I'm not aware of? I.e. is it bad practice to use one user-facing function in another user-facing function?

Reply:
`get_return_value`, `get_return_prob`, `get_return_period` are the function exposed to the user and have the same inputs except for the variable to be returned but different names and docstrings. This is a better way to present to the user rather than one function that does different things (that is what happens behind the scenes). This is a good compression of redundant code done here as it moves that code to a single internal function. Does this makes  sense?

Reply:
Yeah agree with Eric here. I think my discomfort with the other suggestion is also that 'get_return_value' wouldn't be a good name for the shared function, because it actually can do more than that. So I also think it's important for the function names to make sense for what they do (with `_get_return_variable` being the generic one that can do all three)

Reply:
Sure, sounds good to me. Makes sense now, sorry for my initial hesitation. Thanks for the explanation!